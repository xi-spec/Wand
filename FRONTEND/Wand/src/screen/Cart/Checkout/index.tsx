import React from 'react';
import { Text, View, TouchableOpacity, Image } from 'react-native';
import styles from './styles';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { updateStockApi } from '../../../redux/actions/productActions';
import { paymentSuccessfuly } from '../../../redux/actions/userActions';
import { ShoppingCart, User } from '../../../models/initiaState';
import { Icon } from 'react-native-elements';
import { IMAGES } from '../../../constants/images';

function Checkout ({ navigation, cart, user, actions }:
  {cart:ShoppingCart, actions:any, user:User, navigation:{navigate:Function}}) {
  const { cardNumber } = user.card || { undefined };

  function confirmAndPay (user, cart) {
    navigation.navigate('Payment successful');
    actions.updateStockApi(cart);
    actions.paymentSuccessfuly(user, cart);
  }

  return (
    <View style={styles.container}>
    <View style={styles.shippingContainer}>
      <View style={styles.rowContainer}>
        <Text style={styles.titleText}>Shipping information</Text>
        <TouchableOpacity
        testID='navigation'
        onPress={() => navigation.navigate('Address')}>
        <Text style={styles.edit}>edit</Text>
        </TouchableOpacity>
      </View>
        <View style={styles.shadowBox}>
      <View style={styles.iconContainer}>
          <Icon style={styles.icon} name='person-outline' type='ionicon' />
          <Text style={styles.text}>{user.name}</Text>
      </View>

      <View style={styles.iconContainer}>
          <Icon style={styles.icon}name='map-outline' type='ionicon' />
          {user.address
            ? <Text style={styles.text}>{`${user.address.street} ${user.address.city} ${user.address.postcode} ${user.address.country}`}</Text>
            : <Text style={styles.text}>Please enter your shipping address </Text>
           }
      </View>
        </View>
    </View>

    <View style={styles.shippingContainer}>
      <View style={styles.rowContainer}>
        <Text style={styles.titleText}>Payment Method</Text>
        <TouchableOpacity
        testID='navigation'
        onPress={() => navigation.navigate('Payment')}>
        <Text style={styles.edit}>edit</Text>
        </TouchableOpacity>
      </View>
        <View style={styles.shadowBox}>
      <View style={styles.iconContainer}>
        <Image style={styles.card}source={{ uri: IMAGES.card }}></Image>
          {cardNumber
            ? <Text style={styles.cardNumber }>{`**** **** **** ${cardNumber.slice(cardNumber.length - 4)}`}</Text>
            : <Text style={styles.cardNumber }>Please enter your card information</Text>
          }
      </View>
        </View>
    </View>

    <View style={styles.checkoutContainer}>
          <View style={styles.totalContainer}>
          <Text style={styles.totalText}>Total:</Text>
          <Text style={styles.totalText}>{`${cart.totalCost.toFixed(2)} euros`}</Text>
          </View>

          <TouchableOpacity
          disabled={!user.address || !user.card}
          testID='pay'
          onPress={() => confirmAndPay(user, cart)}
           style={styles.checkout}>
            <Text style={styles.checkoutText}>CONFIRM AND PAY</Text>
          </TouchableOpacity>
        </View>

    </View>

  );
}

function mapStateToProps ({ cart, user }) {
  return {
    cart,
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      paymentSuccessfuly,
      updateStockApi
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Checkout);
