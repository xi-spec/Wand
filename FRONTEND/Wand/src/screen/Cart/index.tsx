import React from 'react';
import { Text, TouchableOpacity, Image, FlatList, View } from 'react-native';
import styles from './styles';
import { Icon } from 'react-native-elements';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  increaseQuantityCart,
  decreaseQuantityCart,
  deleteProductCart
} from '../../redux/actions/productActions';
import EmptyCart from './EmptyCart';
import { ShoppingCart, User } from '../../models/initiaState';
import { COLORS } from '../../constants/theme';

function Cart ({ navigation, cart, actions, user }
  :{cart:ShoppingCart, actions:any,
    navigation:{navigate:Function, reset:Function},
    user:User}) {
  return (
    cart.totalCost === 0
      ? <EmptyCart navigation={navigation}/>
      : <View style={styles.container}>
           <FlatList
           bounces={false}
           style={styles.list}
           data={cart.cartList}
           numColumns={1}
           keyExtractor={(item) => item._id}
           showsVerticalScrollIndicator={false}
           renderItem={({ item }) =>
           <View style={styles.shadowBox}>
           <View style={styles.cartContainer}>
             <TouchableOpacity
             style={styles.imageContainer}
             key={item.name}
             testID='productImage'
             onPress={() => navigation.navigate('Detail', { productId: item._id })}>
           <Image source={{ uri: item.images.alls[0] }}
           style={styles.image} />
             </TouchableOpacity>
             <View style={styles.textContainer}>
               <View style={styles.titleContainer}>
             <Text style={styles.detail}>{item.name}</Text>
             <TouchableOpacity
           onPress={() => actions.deleteProductCart(item._id, item.quantity)}
           testID='delete'
           >
             <Icon name='close-outline' type="ionicon"></Icon>
           </TouchableOpacity>
               </View>
             <Text style={styles.motto}>{item.motto ? item.motto : 'What a Nice Day!' }</Text>

             <View style={styles.rightContainer}>
             <Text style={{ ...styles.detail, ...styles.cost }}>{`${item.cost.value.toFixed(2)} ${item.cost.currency}`}</Text>
             <View style={styles.chevronContainer}>
               <TouchableOpacity
               testID='decrease'
               disabled={item.quantity === 1}
               onPress={() => actions.decreaseQuantityCart(item._id, item.stock)}>
             <Icon style={styles.iconLeft} name="remove-circle-outline" type='ionicon' size={20} color={COLORS.primary}/>
               </TouchableOpacity>
             <Text style={{ ...styles.detail, ...styles.quantity }}>{item.quantity}</Text>
             <TouchableOpacity
                  testID='increase'
             onPress={() => actions.increaseQuantityCart(item._id, item.stock)}
             disabled={!item.stock}
             >
             <Icon style={styles.iconRight} name="add-circle-outline" type='ionicon' size={20} color={COLORS.primary}/>
             </TouchableOpacity>

             </View>
             </View>
             </View>
         <View>
         </View>
           </View>
           </View>

             }>
        </FlatList>
        <View style={styles.checkoutContainer}>
          <View style={styles.totalContainer}>
          <Text style={styles.totalText}>Total:</Text>
          <Text style={styles.totalText}>{`${cart.totalCost.toFixed(2)} euros`}</Text>
          </View>

          <TouchableOpacity
          testID='checkout'
          onPress={() => user.isLogged ? navigation.navigate('Checkout') : navigation.navigate('Profile')}
           style={styles.checkout}>
            <Text style={styles.checkoutText}>CHECKOUT</Text>
          </TouchableOpacity>
        </View>

        </View>

  );
}

function mapStateToProps ({ cart, user }) {
  return {
    cart,
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      increaseQuantityCart,
      decreaseQuantityCart,
      deleteProductCart
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Cart);
