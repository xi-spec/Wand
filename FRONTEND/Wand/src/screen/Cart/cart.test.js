import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/productActions';
import { Provider } from 'react-redux';
import Cart from './index';

describe('Given a Cart component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    cart: {
      cartList: [{
        _id: '1',
        images: { alls: ['url'] },
        cost: { value: 1, currency: 'euro' }
      }],
      totalItem: 0,
      totalCost: 1
    }
  });
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'increaseQuantityCart').mockReturnValueOnce({ type: '' });
    jest.spyOn(actions, 'decreaseQuantityCart').mockReturnValueOnce({ type: '' });
    jest.spyOn(actions, 'deleteProductCart').mockReturnValueOnce({ type: '' });

    navigation = {
      navigate: jest.fn()
    };
  });

  describe('When is rendered with cart.totalCost === 0', () => {
    test('Then is matched to snapshop', () => {
      const store = mockStore({
        cart: {
          cartList: [],
          totalItem: 0,
          totalCost: 0
        }
      });
      const tree = render(<Provider store={store}><Cart /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When is rendered with cart.totalCost === 1', () => {
    test('Then is mached snapshot', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' }
          }],
          totalItem: 0,
          totalCost: 1
        }
      });
      const tree = render(<Provider store={store}><Cart /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
  describe('When is rendered with item.motto === hello', () => {
    test('Then is mached snapshot', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' },
            motto: 'hello'
          }],
          totalItem: 0,
          totalCost: 1
        }
      });
      const tree = render(<Provider store={store}><Cart /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
  describe('When productImage  is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const navigation = {
        navigate: jest.fn()
      };

      const { getByTestId } = render(
      <Provider store={store}><Cart navigation={navigation} /></Provider>);

      fireEvent.press(getByTestId('productImage'));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When item.quantity is > 1  and decrease quantity button is pressed ', () => {
    test('Then actions.decreaseQuantityCart is invoked', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' },
            quantity: 2
          }],
          totalCost: 1

        }
      });

      const { getByTestId } = render(
      <Provider store={store}><Cart/></Provider>);

      fireEvent.press(getByTestId('decrease'));

      expect(actions.decreaseQuantityCart).toHaveBeenCalled();
    });
  });

  describe('When have item.stock  and increase quantity button is pressed', () => {
    test('Then actions.increaseQuantityCart is invoked', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' },
            quantity: 2,
            stock: 1
          }],
          totalCost: 1
        }
      });

      const { getByTestId } = render(
      <Provider store={store}><Cart/></Provider>);

      fireEvent.press(getByTestId('increase'));

      expect(actions.increaseQuantityCart).toHaveBeenCalled();
    });
  });

  describe('When delete button is pressed', () => {
    test('Then actions.deleteProductCart is invoked', () => {
      const { getByTestId } = render(
      <Provider store={store}><Cart/></Provider>);

      fireEvent.press(getByTestId('delete'));

      expect(actions.deleteProductCart).toHaveBeenCalled();
    });
  });
  describe('When checkout button is pressed', () => {
    test('Then navigation.navigate is invoked with Checkout when user isLogged', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' },
            quantity: 2,
            stock: 1
          }],
          totalCost: 1
        },
        user: {
          isLogged: true
        }
      });

      const { getByTestId } = render(
      <Provider store={store}><Cart navigation={navigation}/></Provider>);

      fireEvent.press(getByTestId('checkout'));

      expect(navigation.navigate).toHaveBeenCalled();
    });

    test('Then navigation.navigate is invoked with Checkout when user isLogged', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' },
            quantity: 2,
            stock: 1
          }],
          totalCost: 1
        },
        user: {
          isLogged: false
        }
      });
      const { getByTestId } = render(
      <Provider store={store}><Cart navigation={navigation}/></Provider>);

      fireEvent.press(getByTestId('checkout'));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
