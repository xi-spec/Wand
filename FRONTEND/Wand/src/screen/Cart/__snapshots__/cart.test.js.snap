// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Cart component When is rendered with cart.totalCost === 0 Then is matched to snapshop 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 266.8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://trello-attachments.s3.amazonaws.com/603be1b68ce9c654d30ae543/60506d7b3b22568450a0dc85/e2f00d68e8c7e38709c459cd55d470ad/finall3-e1604082541825-400x400.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    YOUR CART IS EMPTY
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    Check if there any products in your wishlist and snatch them up before they're gone
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 266.8,
        "height": 80,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E3192",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="shopNow"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        SHOP NOW
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="favourite"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        VIEW WISHLIST
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a Cart component When is rendered with cart.totalCost === 1 Then is mached snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "_id": "1",
          "cost": Object {
            "currency": "euro",
            "value": 1,
          },
          "images": Object {
            "alls": Array [
              "url",
            ],
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginBottom": 100,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "elevation": 5,
              "height": 200,
              "justifyContent": "center",
              "marginTop": 10,
              "shadowColor": "grey",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "margin": 5,
                  "opacity": 1,
                  "paddingLeft": 10,
                }
              }
              testID="productImage"
            >
              <Image
                source={
                  Object {
                    "uri": "url",
                  }
                }
                style={
                  Object {
                    "borderRadius": 5,
                    "height": 150,
                    "justifyContent": "center",
                    "width": 100,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "width": 250,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingBottom": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontFamily": "RalewayRegular",
                      "fontSize": 18,
                      "marginBottom": 5,
                      "paddingTop": 5,
                    }
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="delete"
                >
                  <View
                    style={
                      Object {
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "fontFamily": "RalewayExtraLight",
                    "fontSize": 14,
                    "marginBottom": 5,
                    "paddingTop": 5,
                  }
                }
              >
                What a Nice Day!
              </Text>
              <View
                style={
                  Object {
                    "marginLeft": 180,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#F3603F",
                      "fontFamily": "MontserratRegular",
                      "fontSize": 18,
                      "marginBottom": 5,
                      "paddingTop": 15,
                      "width": 200,
                    }
                  }
                >
                  1.00 euro
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "stretch",
                      "flexDirection": "row",
                      "paddingTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="decrease"
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginRight": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#2E3192",
                        "fontFamily": "RalewayRegular",
                        "fontSize": 18,
                        "marginBottom": 5,
                        "marginHorizontal": 5,
                        "paddingTop": 0.5,
                      }
                    }
                  />
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="increase"
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginLeft": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 10,
        "height": 80,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        Total:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        1.00 euros
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="checkout"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        CHECKOUT
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a Cart component When is rendered with item.motto === hello Then is mached snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "_id": "1",
          "cost": Object {
            "currency": "euro",
            "value": 1,
          },
          "images": Object {
            "alls": Array [
              "url",
            ],
          },
          "motto": "hello",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginBottom": 100,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "elevation": 5,
              "height": 200,
              "justifyContent": "center",
              "marginTop": 10,
              "shadowColor": "grey",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "margin": 5,
                  "opacity": 1,
                  "paddingLeft": 10,
                }
              }
              testID="productImage"
            >
              <Image
                source={
                  Object {
                    "uri": "url",
                  }
                }
                style={
                  Object {
                    "borderRadius": 5,
                    "height": 150,
                    "justifyContent": "center",
                    "width": 100,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "height": 150,
                  "width": 250,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingBottom": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontFamily": "RalewayRegular",
                      "fontSize": 18,
                      "marginBottom": 5,
                      "paddingTop": 5,
                    }
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="delete"
                >
                  <View
                    style={
                      Object {
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#242424",
                                "fontSize": 24,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="iconIcon"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "fontFamily": "RalewayExtraLight",
                    "fontSize": 14,
                    "marginBottom": 5,
                    "paddingTop": 5,
                  }
                }
              >
                hello
              </Text>
              <View
                style={
                  Object {
                    "marginLeft": 180,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#F3603F",
                      "fontFamily": "MontserratRegular",
                      "fontSize": 18,
                      "marginBottom": 5,
                      "paddingTop": 15,
                      "width": 200,
                    }
                  }
                >
                  1.00 euro
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "stretch",
                      "flexDirection": "row",
                      "paddingTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="decrease"
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginRight": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#2E3192",
                                  "fontSize": 20,
                                },
                                Object {
                                  "backgroundColor": "transparent",
                                },
                                Object {
                                  "fontFamily": "ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                            testID="iconIcon"
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#2E3192",
                        "fontFamily": "RalewayRegular",
                        "fontSize": 18,
                        "marginBottom": 5,
                        "marginHorizontal": 5,
                        "paddingTop": 0.5,
                      }
                    }
                  />
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="increase"
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginLeft": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#2E3192",
                                  "fontSize": 20,
                                },
                                Object {
                                  "backgroundColor": "transparent",
                                },
                                Object {
                                  "fontFamily": "ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                            testID="iconIcon"
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 10,
        "height": 80,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        Total:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        1.00 euros
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="checkout"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        CHECKOUT
      </Text>
    </View>
  </View>
</View>
`;
